cmd/snap/cmd_get.go

```diff
@@ -96,9 +148,33 @@ func (x *cmdGet) Execute(args []string) error {
 		return err
 	}
 
+	isTerminal := terminal.IsTerminal(int(os.Stdin.Fd()))
+
 	var confToPrint interface{} = conf
-	if !x.Document && len(confKeys) == 1 {
-		confToPrint = conf[confKeys[0]]
+	if !x.Document && !x.List && len(confKeys) == 1 {
+		// if single key was requested, then just output the value unless it's a map,
+		// in which case it will be printed as a list below.
+		if _, ok := conf[confKeys[0]].(map[string]interface{}); !ok {
+			confToPrint = conf[confKeys[0]]
+		}
+	}
+
+	if cfg, ok := confToPrint.(map[string]interface{}); ok && !x.Document {
+		// TODO: remove this conditional and the warning below after a transition period.
+		if isTerminal || x.List {
+			w := tabWriter()
+			defer w.Flush()
+
+			rootRequested := len(confKeys) == 0
```

Comment:
The fact root was requested does not seem relevant here, which perhaps means there's a bug lurking here. Or perhaps I'm missing something as well.

Let's think about it this way: what's the difference between requesting "foo" and getting _{"foo": 123},_ and requesting {} and getting _{"foo": 123}._ Those are both the same documents, right?  Why does it make a difference in presentation?

The fact the flattenConfig has almost exactly the same logic nested twice inside one another is another hint. It sounds like a single loop should work both for root and for anything else.

Or am I'm missing something as well. :)

Reply:
This is true for simple scalar values, but not for complex config values (maps).
Consider config key `"foo"` with value `{"bar":123,"baz":456}`. Whether you request "foo" or the root, the document is the same: `{"foo": {"bar":123,"baz":456}`. But the presentation (produced by `flattenConfig`) needs to be different:

1. 

 if you requested `foo` specifically, the output will be:
`Key        Value
foo.bar    123
foo.baz    456

2. 

if you requested root document, the output will be:
`Key        Value
foo         {...}
`
The inner loop of the main `flattenConfig` loop deals with "expanding" of dicts keys for case #1.

Reply:
Yes, your understanding is correct. I've added one more unit test case to cover this and make it clear. I believe this is what we want in the light of the earlier forum discussions.